openlibrary/plugins/openlibrary/code.py

```diff
@@ -1063,6 +1064,43 @@ def GET(self):
                 args[0], args[1]
             )
             partial = {"partials": str(macro)}
+        elif component == 'SearchFacets':
+            data = json.loads(i.data)
+            path = data.get('path')
+            query = data.get('query', '')
+            parsed_qs = parse_qs(query.replace('?', ''))
+            param = data.get('param', {})
+
+            sort = None
+            search_response = do_search(
+                param, sort, rows=0, spellcheck_count=3, facet=True
+            )
+
+            sidebar = render_template(
+                'search/work_search_facets',
+                param,
+                facet_counts=search_response.facet_counts,
+                async_load=False,
+                path=path,
+                query=parsed_qs,
+            )
+
+            active_facets = render_template(
+                'search/work_search_selected_facets',
+                param,
+                search_response,
+                param.get('q', ''),
+                path=path,
+                query=parsed_qs,
+            )
+
+            partial = {
+                "sidebar": str(sidebar),
+                "title": active_facets.title,
+            }
+
+            if active_facets_markup := str(active_facets).strip():
+                partial["activeFacets"] = active_facets_markup
```

Comment:
Why is only this one wrapped in an `if` statement? Can we put it up above?
```suggestion
                "activeFacets": str(active_facets).strip()
            }
```